## Cleaning Dataset

**Find The File Name**

In [1]:
import os

for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))
        

/kaggle/input/e-commerce-sales-data/E-commerce sales data 2024.csv


**Load the Dataset**

In [2]:
import pandas as pd

df = pd.read_csv('/kaggle/input/e-commerce-sales-data/E-commerce sales data 2024.csv')

df.head()

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


,user id,product id,Interaction type,Time stamp,Unnamed: 4
0,1.0,4c69b61db1fc16e7013b43fc926e502d,purchase,10/10/2023 8:00,NaN
1,2.0,66d49bbed043f5be260fa9f7fbff5957,view,11/10/2023 8:00,NaN
2,3.0,2c55cae269aebf53838484b0d7dd931a,like,12/10/2023 8:00,NaN
3,4.0,18018b6bc416dab347b1b7db79994afa,view,13/10/2023 8:00,NaN
4,5.0,e04b990e95bf73bbe6a3fa09785d7cd0,like,14/10/2023 8:00,NaN


**Inspect The Dataset**

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3294 entries, 0 to 3293
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   user id           2999 non-null   float64
 1   product id        2999 non-null   object 
 2   Interaction type  2871 non-null   object 
 3   Time stamp        2999 non-null   object 
 4   Unnamed: 4        0 non-null      float64
dtypes: float64(2), object(3)
memory usage: 128.8+ KB


**Check Missing Values**

In [4]:
df.isnull().sum()

user id              295
product id           295
Interaction type     423
Time stamp           295
Unnamed: 4          3294
dtype: int64

**Remove the Empty Column**

In [5]:
df = df.drop(columns=['Unnamed: 4'])

**Verify**

In [6]:
df.columns

Index(['user id', 'product id', 'Interaction type', 'Time stamp'], dtype='object')

**Check for Duplicate Rows**

In [7]:
df.duplicated().sum()

294

**Save Progress**

In [8]:
df.to_csv('cleaning_checkpoint.csv', index=False)

**Remove Duplicates**

In [9]:
df = df.drop_duplicates()

**Verify**

In [10]:
df.duplicated().sum()

0

**Check Missing Values Again**

In [11]:
df.isnull().sum()

user id               1
product id            1
Interaction type    129
Time stamp            1
dtype: int64

**Remove Remaining Missing Values**

In [12]:
df = df.dropna()

**Verify**

In [13]:
df.isnull().sum()

user id             0
product id          0
Interaction type    0
Time stamp          0
dtype: int64

**Check Dataset Shape**

In [14]:
df.shape

(2871, 4)

**Standardize Text Values**

In [15]:
df['Interaction type'].value_counts()

Interaction type
like        1145
view         871
purchase     855
Name: count, dtype: int64

In [16]:
df.shape

(2871, 4)

**Save the Final Clean Dataset**

In [17]:
df.to_csv('user_engagement_cleaned.csv', index=False)

## Data Cleaning Summary

- Removed empty column (Unnamed: 4)
- Removed 294 duplicate records
- Removed rows containing missing values
- Converted Time stamp to datetime format
- Final dataset contains 2,871 records and 4 columns

## EDA (Exploratory Data Analysis)

**Total Unique Users**

In [18]:
df['user id'].nunique()

2871

**Total Unique Products**

In [19]:
df['product id'].nunique()

2871

**Top 10 Products**

In [20]:
df['product id'].value_counts().head(10)

product id
4c69b61db1fc16e7013b43fc926e502d    1
491ca5b2eb12adcdae7e15757036a406    1
66a307d75bb29fa26643e896dad4e314    1
1d8afc2fb1d7d4372d01b816c0a60ea2    1
3374b17fd9111d5d3ffcce6693ba334b    1
e3da0412a75c9a1309fff6fb20e2fb5d    1
b3c9b4a7dfcfb24667130f6be0001e74    1
361891eb1d8c876824eaab718c8375bb    1
6b03e6d0ec1ed0bdc26272953c52f45c    1
70573b0eaddb21db8fd2eafc412f3055    1
Name: count, dtype: int64

**Monthly Activity**

In [21]:
df['Month'] = df['Time stamp'].dt.month_name()

df['Month'].value_counts()

AttributeError: Can only use .dt accessor with datetimelike values

**Time stamp column is no longer being treated as a datetime column**

**Run Code to Fix:**

In [ ]:
df['Time stamp'] = pd.to_datetime(
    df['Time stamp'],
    dayfirst=True
)

**Verify**

In [ ]:
df.dtypes

**Create Month column**

In [ ]:
df['Month'] = df['Time stamp'].dt.month_name()

In [ ]:
df['Month'].value_counts()

**Run Codes After Converting**

In [ ]:
df[['Time stamp', 'Month']].head()

In [ ]:
df['user id'].nunique()

In [ ]:
df['product id'].nunique()

In [ ]:
df['product id'].value_counts().head(10)

In [ ]:
df['Month'].value_counts()

In [ ]:
df['user id'].value_counts().head()

In [ ]:
df['product id'].value_counts().head()

**Save Dataset**

In [ ]:
df.to_csv('user_engagement_cleaned.csv', index=False)